lib/scarpe/widget.rb

```diff
@@ -27,6 +31,10 @@ def dsl_name
         n = name.split("::").last.chomp("Widget")
         n.gsub(/(.)([A-Z])/, '\1_\2').downcase
       end
+
+      def find_by_name(name)
+        widget_classes.detect { |k| k.dsl_name == name.to_s || k.alias_name == name.to_sym }
```

Comment:
Nitpick: maybe use k.alias_name.to_s instead of name.to_sym here, so we're not converting half to string and half to sym randomly?

Reply:
I had that thought. It was `name.to_s` that made me think it was better to convert name. Made me think we dont know if name will be a string or a symbol so better to do the converting on it. It felt weird to convert name in one and the method return value in the other